### PR TITLE
Add support for old SLE11 cryptsetup version

### PR DESCRIPTION
**Pull request by aaannz**

### CODE REVIEW

**Comment by aaannz** [comment]
Diff between SLE11 and generic version:

```diff
> diff -u saltboot.py saltboot-sle11.py 
--- saltboot.py 2020-11-10 15:08:38.012619218 +0100
+++ saltboot-sle11.py   2020-11-10 15:11:02.772166411 +0100
@@ -26,10 +26,6 @@
 
 
 def _lsblk_compat(device):
-    res = __salt__['cmd.run_all']("lsblk -J -p {0}".format(device), output_loglevel='trace')
-    if res['retcode'] == 0:
-        return json.loads(res['stdout'])
-
     res = __salt__['cmd.run_all']("lsblk -n -r -o MOUNTPOINT,FSTYPE,NAME {0}".format(device), output_loglevel='trace')
 
     ret = None
@@ -46,10 +42,6 @@
     return ret
 
 def _findmnt_compat(device):
-    res = __salt__['cmd.run_all']("findmnt -J --submounts {0}".format(device), output_loglevel='trace')
-    if res['retcode'] == 0:
-        return json.loads(res['stdout'])
-
     res = __salt__['cmd.run_all']("findmnt -r -n -o TARGET,SOURCE --submounts {0}".format(device), output_loglevel='trace')
 
     ret = None
@@ -144,9 +136,9 @@
         res = __salt__['file.is_blkdev'](luks_device)
         if res:
             log.debug("_luks_open: already opened: " + luks_device)
-
             return luks_device
-        res = __salt__['cmd.run_all']("echo {0} | cryptsetup open --type luks {1} {2}".format(luks_pass, device, luks_name), python_shell=True)
+        res = __salt__['cmd.run_all']("echo {0} | cryptsetup luksOpen {1} {2}".format(luks_pass, device, luks_name), python_shell=True)
+
         log.debug("_luks_open: {0} res: {1}".format(luks_device, res))
         if res['retcode'] != 0:
             return None
@@ -156,7 +148,7 @@
     return luks_device
 
 def _luks_create(device, luks_pass):
-    res = __salt__['cmd.run_all']("echo {0} | cryptsetup --force-password luksFormat {1}".format(luks_pass, device), python_shell=True)
+    res = __salt__['cmd.run_all']("echo {0} | cryptsetup luksFormat {1}".format(luks_pass, device), python_shell=True)
     log.debug("_luks_create: {0} res: {1}".format(device, res))
     if res['retcode'] != 0:
         return None
@@ -927,10 +919,10 @@
     log.debug("_get_image_version {0} {1} {2}".format(mountpoint, need_umount, device))
     if mountpoint is not None:
         if __salt__['file.file_exists'](os.path.join(mountpoint, 'etc/ImageVersion')):
-            namever = str(__salt__['file.read'](os.path.join(mountpoint, 'etc/ImageVersion'))).rstrip()
+            namever = str(__salt__['cp.get_file_str'](os.path.join(mountpoint, 'etc/ImageVersion'))).rstrip()
         if (namever is not None and
             __salt__['file.file_exists'](os.path.join(mountpoint, 'etc/ImageVersion-' + namever))):
-            h = str(__salt__['file.read'](os.path.join(mountpoint, 'etc/ImageVersion-' + namever))).split()[0]
+            h = str(__salt__['cp.get_file_str'](os.path.join(mountpoint, 'etc/ImageVersion-' + namever))).split()[0]

         if need_umount:
             _try_umount_device(device)
@@ -1184,7 +1176,7 @@

         cache_path = service_mountpoint + urlparse(image['url']).path

-        if __salt__['file.file_exists'](cache_path) and str(__salt__['file.read'](cache_path + '.hash')).rstrip() == image.get('hash'):
+        if __salt__['file.file_exists'](cache_path) and str(__salt__['cp.get_file_str'](cache_path + '.hash')).rstrip() == image.get('hash'):
             ret['comment'] += "Found cached image {0}.\n".format(cache_path)

             if compr:
@@ -1223,7 +1215,7 @@
         hash_exists = __salt__['file.file_exists'](cache_path + '.hash')
         log.debug('check cache path {0}: exists {1}, hash {2}'.format(cache_path, img_exists, hash_exists))
         if img_exists and hash_exists:
-            cache_hash = str(__salt__['file.read'](cache_path + '.hash')).rstrip()
+            cache_hash = str(__salt__['cp.get_file_str'](cache_path + '.hash')).rstrip()
             pillar_hash = image.get('hash')
             log.debug('cache hash "{0}", pillar hash "{1}"'.format(cache_hash, pillar_hash))
             if cache_hash == pillar_hash:
@@ -1751,7 +1743,7 @@
         __salt__['cmd.run_all']("echo -n 'salt_device={0} root={1} {2}' >/update_kernel_cmdline".format(salt_device, root_device['device'], terminal_kernel_parameters), python_shell=True, output_loglevel='trace')

         # adjust grub configuration
-        __salt__['file.replace'](os.path.join(prefix, 'etc/default/grub'), '^GRUB_CMDLINE_LINUX=', "GRUB_CMDLINE_LINUX='{0}'".format(terminal_kernel_parameters), append_if_not_found=True)
+        # __salt__['file.replace'](os.path.join(prefix, 'etc/default/grub'), '^GRUB_CMDLINE_LINUX=', "GRUB_CMDLINE_LINUX='{0}'".format(terminal_kernel_parameters), append_if_not_found=True)

     return ret

@@ -1785,7 +1777,7 @@
     if parameters is None:
         return True, 'OK'

-    kernel_cmdline = str(__salt__['file.read']('/proc/cmdline')).split()
+    kernel_cmdline = str(__salt__['cp.get_file_str']('/proc/cmdline')).split()

     ok = True
     msg = 'Missing kernel parameters: '
```

File: saltboot-formula/saltboot/init.sls [hunk 1]
```diff
@@ -85,40 +90,40 @@ no_pillar:
...
+      - {{saltboot}}: boot_system
 
 boot_system:
-  saltboot.verify_and_boot_system:
+  {{saltboot}}.verify_and_boot_system:
     - partitioning: {{ partitioning|yaml }}
     - images: {{ images|yaml }}
     - boot_images: {{ boot_images|yaml }}
-    - action: {{ salt['pillar.get']('saltboot:kernel_action', 'kexec') }}
+    - action: {{ salt['pillar.get']('saltboot.kernel_action', 'kexec') }}
```
Comment (nadvornik) [inline saltboot-formula/saltboot/init.sls:126]: why this change?

Comment (aaannz) [reply saltboot-formula/saltboot/init.sls:126]: Ah, that is a mistake in seding!

**Review by nadvornik — APPROVED** [review]
LGTM
